src/main/catalog-sources/weblinks.ts

```diff
@@ -19,84 +19,83 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import { observable, reaction } from "mobx";
+import { computed, observable, reaction } from "mobx";
 import { WeblinkStore } from "../../common/weblink-store";
 import { WebLink } from "../../common/catalog-entities";
 import { catalogEntityRegistry } from "../catalog";
 import got from "got";
-import logger from "../logger";
-import { docsUrl, slackUrl } from "../../common/vars";
-
-const defaultLinks = [
-  { title: "Lens Website", url: "https://k8slens.dev" },
-  { title: "Lens Documentation", url: docsUrl },
-  { title: "Lens Community Slack", url: slackUrl },
-  { title: "Kubernetes Documentation", url: "https://kubernetes.io/docs/home/" },
-  { title: "Lens on Twitter", url: "https://twitter.com/k8slens" },
-  { title: "Lens Official Blog", url: "https://medium.com/k8slens" }
-].map((link) => (
-  new WebLink({
-    metadata: {
-      uid: link.url,
-      name: link.title,
-      source: "app",
-      labels: {}
-    },
-    spec: {
-      url: link.url
-    },
-    status: {
-      phase: "available",
-      active: true
-    }
-  })
-));
+import type { Disposer } from "../../common/utils";
 
 async function validateLink(link: WebLink) {
   try {
     const response = await got.get(link.spec.url, {
-      throwHttpErrors: false
+      throwHttpErrors: false,
+      timeout: 20_000,
     });
 
     if (response.statusCode >= 200 && response.statusCode < 500) {
       link.status.phase = "available";
     } else {
       link.status.phase = "unavailable";
     }
-  } catch(error) {
+  } catch {
     link.status.phase = "unavailable";
   }
 }
 
 
 export function syncWeblinks() {
   const weblinkStore = WeblinkStore.getInstance();
-  const weblinks = observable.array(defaultLinks);
+  const webLinkEntities = observable.map<string, [WebLink, Disposer]>();
+
+  function periodicallyCheckLink(link: WebLink): Disposer {
```

Comment:
This looks scary to me. A timer for each weblink? An extension could programmatically generate a lot of links? I think this is outside the scope of the issue anyway

Reply:
By "outside the scope" do you mean you think I should revert this and only check each link once?

Reply:
I guess you want `setInterval`?
I think I would revert it, seems more of a feature. They are running all the time? Or just when the catalog is displayed?

Reply:
I did mean `setInterval` yes. They would be running all the time, I guess I could make it so that there is an initial spread so that there is some smoothness to the checking (and not spikes every 10 minutes).

I would still say it is a bug because the catalog displays the phase and it should show as "unavailable" if it is actually unavailable.

Reply:
How often do weblinks become unavailable? Maybe I'm missing something but I just don't see the value in using cycles to track this.

Reply:
Anytime the website goes down. it would become unavailable.

Reply:
@jakolehm Should we check them periodically or only once per set change (which is what it was before).

Reply:
Jari says that once initially and then every hour or so.